alibi/explainers/similarity/backends/tensorflow/base.py

```diff
@@ -0,0 +1,97 @@
+"""Tensorflow backend for similarity explainers.
+
+Methods unique to the Tensorflow backend are defined here. The interface this class defines syncs with the torch backend
+in order to ensure that the similarity methods only require to match this interface.
+"""
+
+import random
+import os
+from typing import Callable
+
+import numpy as np
+import tensorflow as tf
+import tensorflow.keras as keras
+
+
+class _TensorFlowBackend:
+    device = None
+
+    @staticmethod
+    def get_grads(
+            model: keras.Model,
+            x: tf.Tensor,
+            y: tf.Tensor,
+            loss_fn: Callable[[tf.Tensor, tf.Tensor], tf.Tensor],
+    ) -> np.ndarray:
+        """
+        Computes the gradients of the loss function with respect to the model's parameters for a single training and
+        target pair.
+
+        Parameters:
+        -----------
+        model:
+            The model to compute gradients for.
+        x:
+            The input data point.
+        y:
+            The target data point.
+        loss_fn:
+            The loss function to use.
+
+        Returns:
+        --------
+        grads: np.ndarray
+            The gradients of the loss function with respect to the model's parameters. This is returned as a flattened \
+            array.
+        """
+
+        with tf.device(_TensorFlowBackend.device):
+            with tf.GradientTape() as tape:
+                output = model(x, training=False)
```

Comment:
Maybe add the training as a parameter to the function to allow for a more general use case?

Reply:
Would be good to hear what kind of use cases you had in mind? We can always expose more arguments in the public API, but it's harder to take them away :)